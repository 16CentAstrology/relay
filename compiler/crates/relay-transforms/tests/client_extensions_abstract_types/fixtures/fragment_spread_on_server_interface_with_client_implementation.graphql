query Foo {
  client_type {
    ...MyFragment
  }
}

fragment MyFragment on MaybeNodeInterface {
  name
}

# %extensions%

extend type Query {
  client_type: ClientType
}

type ClientType implements MaybeNodeInterface {
  name: String
}
