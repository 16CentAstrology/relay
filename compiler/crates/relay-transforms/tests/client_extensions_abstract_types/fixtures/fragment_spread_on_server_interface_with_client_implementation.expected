==================================== INPUT ====================================
query Foo {
  client_type {
    ...MyFragment
  }
}

fragment MyFragment on MaybeNodeInterface {
  name
}

# %extensions%

extend type Query {
  client_type: ClientType
}

type ClientType implements MaybeNodeInterface {
  name: String
}
==================================== OUTPUT ===================================
fragment MyFragment on MaybeNodeInterface {
  name
}

query Foo @__ClientExtensionAbstractTypeMetadataDirective
# ClientExtensionAbstractTypeMetadataDirective {
#     abstract_types: [
#         AbstractTypeImplements {
#             name: "__isMaybeNodeInterface",
#             concrete: [
#                 "ClientType",
#             ],
#         },
#     ],
# }
 {
  client_type {
    ...MyFragment
  }
}
